Samples/WidgetAdvSampleCS/Widget1.xaml.cs

```diff
@@ -148,6 +150,14 @@ await WindowBoundsTextBlock.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPr
             });
         }
 
+        private async void Widget_GameBarCompactModeEnabledChanged(XboxGameBarWidget sender, object args)
+        {
+            await GameBarCompactModeEnabledTextBlock.Dispatcher.RunAsync(Windows.UI.Core.CoreDispatcherPriority.Normal, () =>

```

Comment:
I wouldn't think we want an event handler to await here, probably just fire_and_forget and don't make the handler async?

Reply:
I'm a little confused? We don't have the fire and forget notion in C#. We would have to run it async to have the handler updated to the ui thread from the calling thread (if the event is updated from a background thread)? Unless I'm missunderstanding? @lvalentine

Reply:
I just mean that the event handler itself, `Widget_GameBarCompactModeEnabledChanged` probably doesn't need to await it. Usually its best practice not to block in an event handler so that other clients can get notified of the event as well, unless there is some pattern here that I'm not aware of.

Reply:
It's possible the C# event handler logic already does it in parallel, since C# does a lot of magic stuff under the covers.

Reply:
I guess the `async void` return type is kind of special, and makes it so this isn't awaitable by the caller anyway, so nevermind its not an issue here then.

Reply:
Ah I see. From my understanding, `await` here works similar to co_await where it just suspends the execution for the handler and returns to the caller (so it's non-blocking in that way) we could just convert them to Task.Run executions - but just leaning towards the style that's used in the file :)

Reply:
> I guess the `async void` return type is kind of special, and makes it so this isn't awaitable by the caller anyway, so nevermind its not an issue here then.

Yeah that makes sense! Thanks for the sign-off :)